src/main/java/io/snowdrop/reporting/weekreport/DevelopmentReportImpl.java

```diff
@@ -0,0 +1,86 @@
+package io.snowdrop.reporting.weekreport;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.snowdrop.github.reporting.model.PullRequest;
+import io.snowdrop.github.reporting.model.Repository;
+import io.snowdrop.reporting.MarkdownHelper;
+import io.snowdrop.reporting.ReportConstants;
+import io.snowdrop.reporting.model.Issue;
+import net.steppschuh.markdowngenerator.MarkdownSerializationException;
+import net.steppschuh.markdowngenerator.link.Link;
+import net.steppschuh.markdowngenerator.list.UnorderedList;
+import net.steppschuh.markdowngenerator.text.Text;
+import net.steppschuh.markdowngenerator.text.TextBuilder;
+
+/**
+ * <p>Generates the weekly report.</p>
+ */
+public class DevelopmentReportImpl {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DevelopmentReportImpl.class);
+
+    private Set<String> users;
+    private Date startDate = null;
+    private Date endDate = null;
+
+    public DevelopmentReportImpl(final Date pStartDate, final Date pEndDate, final Set<String> pusers) {
+        users = pusers;
+        startDate = pStartDate;
+        endDate = pEndDate;
+    }
+
+    public static DevelopmentReportImpl build(final Date pStartDate, final Date pEndDate, final Set<String> pusers) {
+        return new DevelopmentReportImpl(pStartDate, pEndDate, pusers);
+    }
+
+    /**
+     * @param pByModifiedDate
+     * @return
+     */
+    public String buildWeeklyReport(final List<Repository> pByModifiedDate) {
```

Comment:
Can you describe what this part of the code will do like also the query which is done under the hood ? Maybe having an example will help

Reply:
That is the method that launches the query and then builds the report. I'll add this information to the method comment.